1/app/activities/SignUpActivity.js

```diff
@@ -0,0 +1,227 @@
+import React, { Component } from 'react';
+import {
+    StyleSheet,
+    Button,
+    View,
+    TextInput,
+    Text,
+} from 'react-native';
+import firebase from './src/firebase'
+
+
+
+export default class SignUpActivity extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            email: "",
+            emailValid: true,
+            name: "",
+            nameValid: true,
+            phone: "",
+            phoneValid: true,
+            password: "",
+            passwordValid: true,
+        };
+    }
+
+
+
+
+
+    signUp = async (email, password, name, phone) => {
+        await firebase.auth().createUserWithEmailAndPassword(email, password).then(function () {
+            var user = firebase.auth().currentUser;
+            user.updateProfile({
+                displayName: name,
+                phoneNumber: phone
+            });
+        });
+        await firebase.auth().signInWithEmailAndPassword(email, password);
+        this.props.navigation.navigate('signIn');
+
+    }
+   
+    render() {
+
+        return (
+
+            <View style={styles.container}>
+
+                <Text style={styles.text}>Email</Text>
+              
+                <TextInput
+                    style={styles.inputContainer}
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    onChangeText={email => this.validate(email, "email")}
+                />
+              
+                <Text style={[styles.errorMessage, this.state.emailValid ? styles.setInvisible : null]}>Email is not valid</Text>
```

Comment:
я б таке записав у 3 стрічки
```suggestion
                <Text style={[styles.errorMessage, this.state.emailValid ? styles.setInvisible : null]}>
                    Email is not valid
                </Text>
```